<?xml version="1.0"?>

<launch>
  <arg name="example" default=""/>
  
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find mrpt_rbpf_slam)/config/rosconsole.config"/>
    
  <!--SLAM RUN--> 
  <node pkg="rviz" type="rviz" name="rviz_nav" output="screen" args="-d $(find mrpt_rbpf_slam)/rviz/rviz_conf.rviz"></node>
  <node pkg="mrpt_rbpf_slam" type="mrpt_rbpf_slam" name="mrpt_rbpf_slam" output="screen">
    <param name="rawlog_play_delay" value="0.01"/>
    <param name="ini_filename" value="$(find mrpt_rbpf_slam)/tutorial/grid_slam_demo.ini"/>
    <param name="rawlog_filename" value="$(find mrpt_rbpf_slam)/tutorial/grid_slam_demo$(arg example).rawlog"/>
    <param name="odom_frame_id" value="r1/odom"/>
    <param name="global_frame_id" value="/map"/>
    <param name="base_frame_id" value="r1/base_link"/>
    <!-- for multiple sensors, place comma eg "r1/front_laser/scan,/beacon" -->
    <param name="sensor_source" value="r1/front_laser/scan"/>
    <rosparam command="load" file="$(find mrpt_rbpf_slam)/config/default.yaml" /> 
  </node>
</launch>

